docs/en/docs/encoders.md

```diff
@@ -0,0 +1,140 @@
+# Encoders
+
+Encoders are responsible how to encode responses.
+
+## Default Encoders
+
+In order to understand how to serialise a specific object into `json`, Lilya has some default
+encoders that evaluates when tries to *guess* the response type.
+
+* `DataclassEncoder` - Serialises `dataclass` objects.
+* `NamedTupleEncoder` - Serialises `NamedTuple` objects.
+* `ModelDumpEncoder` - Serialises objects by calling its model_dump method. This allows serializing pydantic objects out of the box.
+* `EnumEncoder` - Serialises `Enum` objects.
+* `PurePathEncoder` - Serializes `PurePath` objects.
+* `DateEncoder` - Serializes date and datetime objects.
+* `StructureEncoder` - Serializes more complex data types. `set, frozenset, GeneratorType, tuple, deque`.
+
+What a brand new encoder is needed and it is not natively supported by Lilya? Well, [building a custom encoder](#build-a-custom-encoder)
+is extremly easy and possible.
+
+## Build a custom encoder
+
+As mentioned before, Lilya has [default encoders](#default-encoders) that are used to transform a response
+into a `json` serialisable response.
+
+To build a custom encoder you must use the `Encoder` class from Lilya and override the `serialize()` function
+where it applies the serialisation process of the encoder type.
+
+Then you **must register the encoder** for Lilya to use it.
+
+When defining an encoder the `__type__` or `def is_type(self, value: Any) -> bool:`
+**must be declared or overridden**.
+
+When the `__type__` is properly declared, the default `is_type` will evaluate the object against the
+type and return `True` or `False`.
+
+This is used internally to understand the type of encoder that will be applied to a given object.
+
+!!! warning
+    If you are not able to provide the `__type__` for any reason and you just want to override the
+    default evaluation process, simple override the `is_type()` and apply your custom logic there.
+
+    E.g.: In Python 3.8, for a Pydantic `BaseModel` if passed in the `__type__`, it will throw an
+    error due to Pydantic internals, so to workaround this issue, you can simply override the `is_type()`
+    and apply the logic that validates the type of the object and returns a boolean.
+
+```python
```

Comment:
All examples of lilya live inside the docs_src, in fact, sometimes we put the examples here but this should be inside a file inside docs_src

Reply:
this was a leftover